### DISCUSSION TITLE
How to map JSON Product Type Definition Types to Xml Category Specific Product Type.

### DISCUSSION THREAD

**Discussion by IQUAYE** [amzn/selling-partner-api-models#2809] [General]
From the moment I realized that the Amazon SP-API does not support all product types, I have been finding ways to list new products using the SP-API POST_PRODUCT_DATA feed.

Currently, I have an implementation that allows my users to specify a particular product type from a list of previously acquired product type definitions. After that, I fetch the corresponding [PRODUCT LISTING] definition schema for my users to fill in both required and optional fields. [I prefer this approach of entry because it's more user-friendly than using direct XML tag names from the various XSD files. i.e it has explanations and examples for what each field is used for unlike in the XML files].

I want to use the information provided in the previous steps to push new content to Amazon using the POST_PRODUCT_DATA feed
by converting the data into a [Product.xsd](https://images-na.ssl-images-amazon.com/images/G/01/rainier/help/xsd/release_4_1/Product.xsd).

All is going so far until I have to tell which Product Type Category is actually the correct one for the JSON Product Type Definition productType that the user selected in step one.

I want to do this based on the assumption that there is a kind of correlation between the Product Type Definitions Types and the Product Types in the [Product.xsd](https://images-na.ssl-images-amazon.com/images/G/01/rainier/help/xsd/release_4_1/Product.xsd) xml.

**Comment by wtom** [comment]
I have the same problem, i want to use the product type "LADDER", but it's not in the Product.xsd file under HomeImprovement.xsd.

**Comment by tom-it** [comment]
pretty confusing to say the least, i have a csv file with category names, which dont map exactly with the xml categories :( for now I will rewrite to use the xsd property names, and add a pretty name by spacing where a capital letter is used, but its not ideal
